parsers/ELEXON.py

```diff
@@ -0,0 +1,202 @@
+                   logger=logging.getLogger(__name__)):
+    if not csv_text:
+        return None
+
+    sorted_zone_keys = sorted([zone_key1, zone_key2])
+
+    exchange = '->'.join(sorted_zone_keys)
+    data_points = list()
+    lines = csv_text.split('\n')
+
+    # check field count in report is as expected
+    field_count = len(lines[1].split(','))
+    if field_count != EXPECTED_FIELDS['INTERFUELHH']:
+        logger.warning(
+            'Expected {} fields in INTERFUELHH report, got {}'.format(
+                EXPECTED_FIELDS['INTERFUELHH'], field_count
+            )
+        )
+
+    for line in lines[1:-1]:
+        fields = line.split(',')
+
+        # create the datetime
+        datetime = datetime_from_date_sp(fields[1], fields[2])
+
+        data = {
+            'sortedZoneKeys': exchange,
+            'datetime': datetime,
+            'source': 'bmreports.com'
+        }
+
+        # positive value implies import to GB
+        multiplier = -1 if 'GB' in sorted_zone_keys[0] else 1
+        data['netFlow'] = float(fields[EXCHANGES[exchange]]) * multiplier
+        data_points.append(data)
+
+    if not target_datetime:
+        return data_points[-1]
+    else:
+        return data_points
+
+
+def parse_production(csv_text, target_datetime=None,
+                     logger=logging.getLogger(__name__)):
+    if not csv_text:
+        return None
+
+    data_points = list()
+    lines = csv_text.split('\n')
+
+    # check field count in report is as expected
+    field_count = len(lines[1].split(','))
+    if field_count != EXPECTED_FIELDS['FUELINST']:
+        logger.warning(
+            'Expected {} fields in FUELINST report, got {}'.format(
+                EXPECTED_FIELDS['FUELINST'], field_count
+            )
+        )
+
+    for line in lines[1:-1]:
+        fields = line.split(',')
+
+        # create the datetime
+        date_obj = dt.datetime.strptime(fields[3], '%Y%m%d%H%M%S')
+        datetime = arrow.get(date_obj).replace(minutes=-5).to('Europe/London')
+
+        data = {
+            'zoneKey': 'GB',
+            'datetime': datetime.datetime,
+            'production': {},
+            'storage': {},
+            'source': 'bmreports.com'
+        }
+
+        # read the fuel values into the datapoint
+        for _type, fuels in FUELS.items():
+            for fuel, field_nums in fuels.items():
+                if not isinstance(field_nums, list):
+                    field_nums = [field_nums, ]
+                data[_type][fuel] = sum([float(fields[i]) for i in field_nums])
+
+        data_points.append(data)
+
+    if not target_datetime:
+        return data_points[-1]
+    else:
+        return data_points
+
+
+def datetime_from_date_sp(date, sp):
+    date = dt.datetime.strptime(date, '%Y%m%d').date()
+    sp = int(sp)
+    hour = int((sp - 1) / 2)
+    minute = 0 if sp % 2 == 1 else 30
+    datetime = arrow.get(dt.datetime.combine(date, dt.time(hour, minute)))
```

Comment:
once you have your date it might be easier just to use `arrow.get(date).shift(minutes=30*sp)` (assuming `sp` is zero-based).
Also it might be more logical to cast `sp` to int outside of the function.